### DISCUSSION TITLE
Flask-RESTful dependence on Flask 2.3 deprecated propagated_exceptions

### DISCUSSION THREAD

**Discussion by transreductionist** [pallets/flask#5019] [Q&A] [answered]
I was hoping to get a sanity check on a fix I would like to propose to the Flask-RESTful code.

There is one line of code in Flask-RESTful that depends upon current_app.propagate_exceptions of Flask. When this is deprecated in Flask 2.3 that line of code will fail. Here is the line of code in the `if` statement:

```
if not isinstance(e, HTTPException) and current_app.propagate_exceptions:
    exc_type, exc_value, tb = sys.exc_info()
    if exc_value is e:
        raise
    else:
        raise e
```

Here is how I think that code can be fixed:

```
propagate_exceptions = self.app.debug or self.app.testing or self.app.config['PROPAGATE_EXCEPTIONS']

if not isinstance(e, HTTPException) and propagate_exceptions:
    exc_type, exc_value, tb = sys.exc_info()
    if exc_value is e:
        raise
    else:
        raise e
```

Regards,

Aaron

**Comment by davidism** [comment, marked as answer]
Looks like an issue with Flask-Restful, so it should be reported there. It already was reported https://github.com/flask-restful/flask-restful/issues/957, looks like you reported a duplicate. We're not the maintainers of Flask-Restful, so asking here is off topic.

**Reply by transreductionist** [reply]
I wasn't referring to Flask. I am putting up a pull request in Flask-RESTful and was interested in feedback from the Flask group for the fix I will implement. From the original question:

```
I was hoping to get a sanity check on a fix I would like to propose to the Flask-RESTful code.
```

I give the Flask-RESTful code that needs to be fixed in the first code block. and in the second code block with the suggested fix in.
